### ISSUE TITLE
Empty source control panel using scm.experimental.showHistoryGraph 

### ISSUE THREAD

**Issue by BriceBoy** [microsoft/vscode#219953]
Does this issue occur when all extensions are disabled?: Yes/No

- VS Code Version: Version : 1.91.0 (user setup)
- OS Version: Windows_NT x64 10.0.22631

Steps to Reproduce:

1. Turn on **scm.experimental.showHistoryGraph**
2. Open another editor
3. Create modifications to populate source control pending modifications
--> Source control shows 'X' modifications pending on icon but is emtpy

**Comment by BriceBoy** [comment]
Moreover : 
- I turned option off to see modifications again
- I then turned it on again
- I could add modifications
- I could add commit message
- I clicked Validation button
--> Validation button is still blue and changes still visible even if commit is already pushed :
![image](https://github.com/microsoft/vscode/assets/45175007/c43ec9fe-f369-48cb-9204-580fd289f54c)

**Comment by VSCodeTriageBot** [comment]
Thanks for reporting this issue! Unfortunately, it's hard for us to understand what issue you're seeing. Please help us out by providing a screen recording showing exactly what isn't working as expected. While we can work with most standard formats, `.gif` files are preferred as they are displayed inline on GitHub. You may find https://gifcap.dev helpful as a browser-based gif recording tool.

If the issue depends on keyboard input, you can help us by enabling screencast mode for the recording (`Developer: Toggle Screencast Mode` in the command palette). Lastly, please attach this file via the GitHub web interface as emailed responses will strip files out from the issue.

Happy coding!

**Comment by BriceBoy** [comment]
I'm not able to reproduce it but if I ever encounter it I'll post a screenshot. I can still make a little photo montage to try explaining better my problem. 
If I can remembere well it was like this (I saw badge notification because I had modifications pending but could not do anything to it as they won't show up in the menu)

![image](https://github.com/microsoft/vscode/assets/45175007/a660ff30-acac-4e66-8c8d-d4d19aa8b875)

**Comment by lszomoru** [comment]
Thanks for the update. If you are able to reproduce the issue again, I would love to get additional information to investigate this. Thanks!

**Comment by rdev32** [comment]
hi there, this is the expected behaviour when you enable `"scm.experimental.showHistoryGraph": true,`

![image](https://github.com/microsoft/vscode/assets/78289548/56102357-22d5-422c-a26c-651ee38752e0)

but as @BriceBoy said.. nothing happens. The file view is still same

**Comment by lszomoru** [comment]
When this happens, can you please check the developer console that you can enable using "Developer: Toggle Developer Tools" command in the command palette. It would also be helpful if you could also share the contents of [the git output window](https://code.visualstudio.com/docs/sourcecontrol/overview#_git-output-window). Thanks!

**Comment by Fydon** [comment]
When using Remote SSH and `scm.experimental.showHistoryGraph` is on, I see a blank section where staged changes and changes should show. Turning `scm.experimental.showHistoryGraph` off immediately results in the panel displaying staged changes and changes.

Turning `scm.experimental.showHistoryGraph` back on does not immediately result in it going blank again, but running Reload Window results in a blank panel again.

**Comment by lszomoru** [comment]
@Fydon, thank you! That is very helpful! Could you please run `git show 90bdbed3a41478cb6974912069938c41a87eb4ec` from the integrated terminal and share the output? Any details that you could share about the `refs/remotes/origin/master`, `refs/heads/production`, `refs/remotes/origin/production` would also be helpful. Thanks!

**Comment by Fydon** [comment]
`git show 90bdbed3a41478cb6974912069938c41a87eb4ec` does result is showing the contents of a change as does `git log --format=%H%n%aN%n%aE%n%at%n%ct%n%P%n%D%n%B -z --shortstat --diff-merges=first-parent 90bdbed3a41478cb6974912069938c41a87eb4ec --topo-order --decorate=full refs/heads/production refs/remotes/origin/production refs/remotes/origin/master`. However `git show 90bdbed3a41478cb6974912069938c41a87eb4ec^..` results in

```
fatal: ambiguous argument '90bdbed3a41478cb6974912069938c41a87eb4ec^..': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
```

`git log --format=%H%n%aN%n%aE%n%at%n%ct%n%P%n%D%n%B -z --shortstat --diff-merges=first-parent 90bdbed3a41478cb6974912069938c41a87eb4ec^.. --topo-order --decorate=full refs/heads/production refs/remotes/origin/production refs/remotes/origin/master` results in

```
fatal: ambiguous argument '90bdbed3a41478cb6974912069938c41a87eb4ec^..': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
```

**Comment by BriceBoy** [comment]
I have the problem again, more details as requested : 
![image](https://github.com/microsoft/vscode/assets/45175007/a56932ec-9f41-438a-b8cf-8de770d9c5c6)

![image](https://github.com/microsoft/vscode/assets/45175007/4cd3d9cd-3398-45bd-ae46-2bd06e0a7dbe)

![image](https://github.com/microsoft/vscode/assets/45175007/4ce3c52c-8e20-46b0-abeb-d581f47c4390)

**Comment by lszomoru** [comment]
@Fydon, could you please run the following (should list all parents of this commit):
```
git log -n 1 --pretty=%P 90bdbed3a41478cb6974912069938c41a87eb4ec
``` 

@BriceBoy, could you please run the following
```
git log -n 1 --pretty=%P 2a44fd9a
```

These commits do not seem to have parents. Are they the first commits in the repository? 
If not, what were the steps that you have taken to get into this state? Thank you!

**Comment by BriceBoy** [comment]
I don't think it will help because I'm not in this state anymore. I made some manual commits since then.

I tried but it doesn't show anything

**Comment by lszomoru** [comment]
@Fydon when you say branch, are you talking about the `refs/heads/production`? So this is the first commit on this branch? How was the branch created? Is is expected that the common ancestor between `refs/heads/production`, `refs/remotes/origin/production`, and `refs/remotes/origin/master` is the first commit in the repository?

**Comment by Fydon** [comment]
Yes `90bdbed3a41478cb6974912069938c41a87eb4ec` is the first commit in `refs/heads/production`. It was created before I worked with the repository so I can only guess how it was made. `refs/remotes/origin/production` is a `git push` of `refs/heads/production` to `origin`.

`refs/heads/master` is a branch that only has with a single commit of `90bdbed3a41478cb6974912069938c41a87eb4ec`. It seems that `refs/remotes/origin/master` has a commit similar to `90bdbed3a41478cb6974912069938c41a87eb4ec` but with a different hash. It is showing that `refs/remotes/origin/master` has 1 commit to pull and 1 commit to push, so maybe something like a rebase happened to the commit.

In my case the branch should just be deleted, but I'll leave it so that this can be debugged.

**Comment by lszomoru** [comment]
This issue has been fixed in the latest VS Code Insiders release (version information below). The fix will also be included in the next VS Code Stable recovery release that is scheduled to be released later this week. 

```
Version: 1.92.0-insider
Commit: 8703a11164db333db471576d38ddc143d6ad7ead
Date: 2024-07-10T08:26:11.347Z
Electron: 30.1.2
ElectronBuildId: 9759760
Chromium: 124.0.6367.243
Node.js: 20.14.0
V8: 12.4.254.20-electron.0
OS: Darwin arm64 23.5.0
```

**Comment by Fydon** [comment]
I can confirm its working with that version

**Comment by Fydon** [comment]
I can confirm that the incoming/outgoing graph is no longer empty, that staged changes and changes display.

The graph appears to be showing all the commits in `refs/heads/production`. Is it assuming that `refs/remotes/origin/master` will receive all the changes, even though it will only receive the single commit in `refs/remotes/origin/master`? Should I open a new issue, given that the problem with the source control panel being empty is fixed?

**Comment by lszomoru** [comment]
@Fydon, yes please file a separate issue so that I can better understand the exact structure of your repository. Thanks!

**Comment by Fydon** [comment]
Sorry reading the [description in release notes](https://code.visualstudio.com/updates/v1_91#_incomingoutgoing-changes-graph-preview) again, I misunderstood what the new functionality is trying to do. If your changes have a base branch, its showing all the changes in both branches since the branch was created, which may also include changes that need to be pulled or pushed. This is working as expected. I thought that it was trying to only show incoming and outgoing changes, but I see now it's trying to achieve more than that. Sorry.

**Comment by lszomoru** [comment]
@Fydon, thank you very much for reading the release notes to better understand the feature. Yes, the idea here is to show commits related to your inner loop which includes incoming/outgoing as well as commits that have not been merged into the base branch in case one exists. I am currently working on more options to be able to "hide" commits from particular branches (ex: base and/or remote). 

Please continue to use the feature and share your feedback as it is extremely valuable.

**Comment by bvandevliet** [comment]
Maybe this git graph could also be the base for #179053 ??

**Comment by bendtherules** [comment]
Feedback/criticism: Don't like this feature at all. It looks very busy, distracting and takes focus away from the actual changes.  I understand that it's useful without using Gitlens, but why is it so in the face?

Q: Why is it not within a collapsible section? I can't even hide it by collapsing (like the Staged changed above) if I don't like it.
Q: Why is it so hard to disable it? I should be just able to right click and disable sections (just like Gitlens does it). It's not easy to find out what's it called in settings and disable it.
Q: Why does the graph show main branch as well? If i deselect main branch in the hamburger menu, that doesn't seem to persist for long.
